backend/src/appointment/utils.py

```diff
@@ -60,28 +60,33 @@ def decrypt(value):
 
 
 def retrieve_user_url_data(url):
-    """URL Decodes, and retrieves username, signature, and main url from /<username>/<signature>/"""
+    """URL Decodes, and retrieves username, slug/signature, and main url from /<username>/<slug/signature?>/"""
     parsed_url = parse.urlparse(url)
     split_path = [x for x in parsed_url.path.split('/') if x]
-
-    if split_path is None or len(split_path) == 0:
+    
+    # Check for general validity of the path
+    if split_path is None or len(split_path) == 0 or len(split_path) > 3:
         return False
-    # If we have more than two entries, grab the last two
-    elif len(split_path) > 2:
-        split_path = split_path[-2:]
+
+    # Normalize short and long urls to only username and slug/signature (remove the /user/ segment)
+    # FIXME: Handle edge case: A user with username='user' might make trouble here
+    if len(split_path) > 1 and split_path[0] == 'user':
+        split_path.pop(0)
```

Comment:
I didn't find a better way to handle that. The problem is, that we might have a `/user/` segment or might not.

Reply:
We should create a list of "unavailable" usernames. Like user, username, user_name, admin, thunderbird, etc...

Reply:
Agreed! Added https://github.com/thunderbird/appointment/issues/1004 for this